### DISCUSSION TITLE
using batch.from_data_list, I got an AttributeError: 'listBatch' object has no attribute 'stores_as'

### DISCUSSION THREAD

**Discussion by guojy18** [pyg-team/pytorch_geometric#9488] [Q&A]
File "d:\software\anaconda\Anaconda3_2020\envs\ddpgstarlink\lib\site-packages\torch_geometric\data\batch.py", line 97, in from_data_list
batch, slice_dict, inc_dict = collate(
File "d:\software\anaconda\Anaconda3_2020\envs\ddpgstarlink\lib\site-packages\torch_geometric\data\collate.py", line 61, in collate
out.stores_as(data_list[0]) # type: ignore
AttributeError: 'listBatch' object has no attribute 'stores_as'
my environment is pytorch 2.2.0, python3.8, what is wrong

**Comment by akihironitta** [comment]
Would you mind sharing your minimal code and env details to reproduce it?

**Reply by guojy18** [reply]
`experiences.state0` is like [[Data(x,edge_index,edge_weight)],[Data(x,edge_index,edge_weight)],...,[Data(x,edge_index,edge_weight)]]
`for e in experiences:`
`        state0_batch.append(e.state0[0])`
`state0_batch = Batch.from_data_list(state0_batch)`
my environment is python 3.8.19, pytorch2.2.0 (py3.8_cuda11.8_cudnn8_0), pyg2.5.2 (py38_torch_2.2.0_cu118), I use conda install to build it

**Reply by rusty1s** [reply]
It looks like you need to do
```
state0_batch.append(e.state0[0][0])
